src/routes/(app)/admin[[hash=admin_hash]]/pos-payments/+page.server.ts

```diff
@@ -0,0 +1,174 @@
+		},
+		{
+			processor: 'sumup' as const,
+			available: isSumupEnabled(),
+			displayName: 'SumUp'
+		}
+	];
+
+	return {
+		subtypes,
+		availableProcessors
+	};
+};
+
+export const actions: Actions = {
+	create: async ({ request }) => {
+		const formData = await request.formData();
+
+		const parsed = z
+			.object({
+				slug: z
+					.string()
+					.min(1)
+					.regex(/^[a-z0-9-]+$/, 'Slug must be lowercase alphanumeric with hyphens'),
+				name: z.string().min(1),
+				description: z.string().optional(),
+				tapToPayProcessor: z.enum(['', 'stripe', 'sumup']),
+				tapToPayUrl: z.string().optional()
+			})
+			.safeParse({
+				slug: formData.get('slug'),
+				name: formData.get('name'),
+				description: formData.get('description'),
+				tapToPayProcessor: formData.get('tapToPayProcessor') || '',
+				tapToPayUrl: formData.get('tapToPayUrl')
+			});
+
+		if (!parsed.success) {
+			return fail(400, { error: parsed.error.message });
+		}
+
+		const existing = await collections.posPaymentSubtypes.findOne({ slug: parsed.data.slug });
+		if (existing) {
+			return fail(400, { error: 'Subtype with this slug already exists' });
+		}
+
+		const maxSortOrder = await collections.posPaymentSubtypes
+			.find({})
+			.sort({ sortOrder: -1 })
+			.limit(1)
+			.toArray();
+
+		const newSubtype: PosPaymentSubtype = {
+			_id: new ObjectId(),
+			slug: parsed.data.slug,
+			name: parsed.data.name,
+			description: parsed.data.description || undefined,
+			tapToPay: parsed.data.tapToPayProcessor
+				? {
+						processor: parsed.data.tapToPayProcessor as PaymentProcessor,
+						onActivationUrl: parsed.data.tapToPayUrl || undefined
+				  }
+				: undefined,
+			sortOrder: (maxSortOrder[0]?.sortOrder ?? 0) + 1,
+			createdAt: new Date(),
+			updatedAt: new Date()
+		};
+
+		await collections.posPaymentSubtypes.insertOne(newSubtype);
+
+		return { success: true };
+	},
+
+	update: async ({ request }) => {
+		const formData = await request.formData();
+
+		const parsed = z
+			.object({
+				id: z.string().min(1),
+				name: z.string().min(1),
+				description: z.string().optional(),
+				tapToPayProcessor: z.enum(['', 'stripe', 'sumup']),
+				tapToPayUrl: z.string().optional(),
+				disabled: z.boolean().optional()
+			})
+			.safeParse({
+				id: formData.get('id'),
+				name: formData.get('name'),
+				description: formData.get('description'),
+				tapToPayProcessor: formData.get('tapToPayProcessor') || '',
+				tapToPayUrl: formData.get('tapToPayUrl'),
+				disabled: formData.get('disabled') === 'true'
+			});
+
+		if (!parsed.success) {
+			return fail(400, { error: parsed.error.message });
+		}
+
+		const updateData: Partial<PosPaymentSubtype> = {
+			name: parsed.data.name,
+			description: parsed.data.description || undefined,
+			tapToPay: parsed.data.tapToPayProcessor
+				? {
+						processor: parsed.data.tapToPayProcessor as PaymentProcessor,
+						onActivationUrl: parsed.data.tapToPayUrl || undefined
+				  }
+				: undefined,
```

Comment:
Instead of excluding “invalid” values and casting, could you write a function (e.g. `intoPaymentProcessor`) with a type guard? Perhaps using `typedInclude`.

See [this example](https://github.com/be-BOP-io-SA/be-BOP/blob/f235c4ee23b7f6444cd5e70c954019dcc04f522c/src/lib/server/stripe.ts#L23).